packages/components/src/lib/hooks/useSquiggle.ts

```diff
@@ -35,8 +61,19 @@ export const useSquiggle = (args: SquiggleArgs) => {
   const { onChange } = args;
 
   useEffect(() => {
-    onChange?.(result.result.tag === "Ok" ? result.result.value : undefined);
+    onChange?.(
+      result.result.tag === "Ok" ? result.result.value : undefined,
+      result.sourceName
+    );
   }, [result, onChange]);
 
+  useEffect(() => {
+    return () => {
+      if (!args.sourceName) args.project.clean(result.sourceName);
+      if (args.project.getSource(importSourceName(result.sourceName)))
+        args.project.clean(result.sourceName);
+    };
+  });
```

Comment:
You haven't passed the dependency array here. I think that means that the cleanup will fire every time: every time `useSquiggle` is called, the previous cleanup function will be called; and then when the react component is destroyed, the last cleanup will be called.

I recommend manually debugging this: add console.logs and try to force the `useSquiggle` to run multiple times, with `code` value unchanged and `code` value updated, and check that it doesn't re-run the execution too often.

Reply:
Hmmm... I think I may have stumbled across this issue when trying to use it in my build, thanks for noticing! I might try and create a test for this to check for proper cleanup.